environments/binary/Dockerfile

```diff
@@ -1,10 +1,18 @@
+FROM golang:onbuild
+
+WORKDIR /go
+COPY *.go /go/
+
+RUN GOOS=linux GOARCH=386 go build -o server .
+
 FROM alpine:3.5
 
+WORKDIR /app
+
 RUN apk update
-RUN apk add coreutils binutils findutils grep
+RUN apk add coreutils binutils findutils grep curl jq
```

Comment:
I don't have a strong opinion on which packages to include or not, and these two seem really useful for many cases. However, this setup (of having the packages hardcoded in the image) is not scalable or user-friendly for that matter.

We could use a build environment here similar to the [golang](https://github.com/fission/fission/tree/master/environments/go/builder) or [python](https://github.com/fission/fission/tree/master/environments/python/builder) environments. The `binary-build` environment could execute a file in the package called `setup.sh` (if present), which could install the necessary package management. 

@justinbarrick if you want to dive deeper into Fission, that would actually be a perfect follow-up PR to this one. 😁 

cc @soamvasani - your opinion on this? If you agree, let's file this as an issue to keep track of it.

Reply:
Okay, I got this mostly setup. The builder will run `setup.sh`, if it exists and I modified `binary/server.go` to support specialize v2 requests and directories.

However, since setup.sh runs in the builder, any packages installed won't persist.

I'm not sure what the correct way to handle that is, since I don't want to cause the packages to be installed.

Reply:
Ey @justinbarrick - awesome work so far :+1: Sorry for the confusion, I was actually expecting this v2 conversion of the binary env to happen in another/next PR - this PR was already good to go. In either case, I would be happy to accept these changes as well.  

The fact that we don't capture changes to filesystem in the builder is indeed an issue for this environment. Not quite sure what the best way is to solve that issue. I see three options here:

1. We probably need a way to wrap the added packages (or changes to the filesystem in general) and add them to the deploy package. The binary environment can then unwrap the changes and update the runtime environment with the changes during specialization.

2. Another option, which might slower but simpler, is to add support for the execution of another script during the specialization phase. That would allow you to directly manipulate the runtime environment - but at a performance cost as the specialization will become more expensive.

3. The simplest but worst option UX-wise is to leave it at this builder and expect the users to work around it. The user could move the downloaded apk packages (like curl) to the deploy package and add those packages manually to the PATH in the runtime. This sounds like a *okay* solution until we have another solution figured out. For a longer term solution I don't really like it; it is not user-friendly, error prone, and requires users to have a relatively deep understanding of how the build process of Fission works.

^ Either of these solutions is completely out of scope for this PR. So let's keep the PR as it is now. Of course, if you want to help out with it, join us on the Slack so we can discuss this thoroughly before committing to a solution. :)

Minor thing: could you rename `setup.sh` to `build.sh`? It is more consist with the naming of build files in the rest of the project. My bad for suggesting it.

Reply:
I think that makes sense, if we go with option 3 for now, we'll come up with a clean workaround eventually.

As for a long term solution, we could potentially achive option 1 by running the builder script in a chroot, copying over the whole directory to the function container, and then running the function in a chroot.

Reply:
I like the chroot suggestion - that is definitely an option worth exploring.